Extensions/NavMeshPathfindingBehavior/JsExtension.js

```diff
@@ -0,0 +1,866 @@
+      .setLabel('Angle offset');
+
+    behaviorProperties
+      .getOrCreate('extraBorder')
+      .setValue(behaviorContent.getDoubleAttribute('extraBorder').toString())
+      .setLabel('Extra border size');
+
+    return behaviorProperties;
+  };
+  // $FlowExpectedError - ignore Flow warning as we're creating a behavior
+  pathfindingBehavior.initializeContent = function (behaviorContent) {
+    behaviorContent.setDoubleAttribute('acceleration', 400);
+    behaviorContent.setDoubleAttribute('maxSpeed', 200);
+    behaviorContent.setDoubleAttribute('angularMaxSpeed', 180);
+    behaviorContent.setBoolAttribute('rotateObject', true);
+    behaviorContent.setDoubleAttribute('angleOffset', 0);
+    behaviorContent.setDoubleAttribute('extraBorder', 0);
+  };
+  const pathfindingBehaviorDeclaration = extension
+    .addBehavior(
+      'NavMeshPathfindingBehavior',
+      _('NavMesh Pathfinding'),
+      'NavMeshPathfindingBehavior',
+      _(
+        'With this behavior, the object will move while avoiding all objects' +
+          'that are flagged as obstacles.'
+      ),
+      '',
+      'CppPlatform/Extensions/AStaricon.png',
+      'NavMeshPathfindingBehavior',
+      pathfindingBehavior,
+      new gd.BehaviorsSharedData()
+    )
+    .setIncludeFile(
+      'Extensions/NavMeshPathfindingBehavior/navmeshpathfindingruntimebehavior.js'
+    )
+    .addIncludeFile('Extensions/NavMeshPathfindingBehavior/A_navmeshall.js')
+    .addIncludeFile(
+      'Extensions/NavMeshPathfindingBehavior/A_navmeshgenerator.js'
+    );
+
+  pathfindingBehaviorDeclaration
+    .addScopedAction(
+      'SetDestination',
+      _('Move to a position'),
+      _('Move the object to a position'),
+      _('Move _PARAM0_ to _PARAM3_;_PARAM4_'),
+      'Movement on the path (NavMesh)',
+      'CppPlatform/Extensions/AStaricon24.png',
+      'CppPlatform/Extensions/AStaricon16.png'
+    )
+    .addParameter('object', _('Object'), '', false)
+    .addParameter(
+      'behavior',
+      _('Behavior'),
+      'NavMeshPathfindingBehavior',
+      false
+    )
+    .addCodeOnlyParameter('currentScene', '')
+    .addParameter('expression', _('Destination X position'))
+    .addParameter('expression', _('Destination Y position'))
+    .getCodeExtraInformation()
+    .setFunctionName('moveTo');
+
+  pathfindingBehaviorDeclaration
+    .addScopedAction(
+      'DrawNavMesh',
+      _('Draw navigation mesh'),
+      _('Draw navigation mesh'),
+      _('Draw navigation mesh used for _PARAM0_ on _PARAM2_'),
+      'Debug (NavMesh)',
+      'CppPlatform/Extensions/AStaricon24.png',
+      'CppPlatform/Extensions/AStaricon16.png'
+    )
+    .addParameter('object', _('Object'), '', false)
+    .addParameter(
+      'behavior',
+      _('Behavior'),
+      'NavMeshPathfindingBehavior',
+      false
+    )
+    //TODO tried Drawer, PrimitiveDrawing::Drawer, ShapePainterObject
+    .addParameter('objectPtr', _('Shape painter'), '', false)
```

Comment:
This is not working because extensions are assumed not to use objects from other extensions, so `addParameter` will prefix the object type by the extension namespace (so `Drawer` would become PrimitiveDrawing::Drawer in the `PrimitiveDrawing` extension, but here it becomes `NavMeshPathfinding::Drawer`)

I don't have a good solution for now.

Reply:
Actually solution will be to use `.setParameterExtraInfo("PrimitiveDrawing::Drawer")` :) 
See #2894

Reply:
Thanks for the review, is the feature branch planned to be merged or should I use it from there?